### PR TITLE
fix(upgrade): Add an improved user error message for removed Kubernetes APIs

### PR DESCRIPTION
**Pull request by hickeyma**
The error message returned from Kubernetes when APIs are removed is not very informative. This PR adds additional information to the user. It covers the current release manifest APIs validation.

Partial #7219

### CODE REVIEW

**Review by Nastradamus — APPROVED** [review]

File: pkg/action/upgrade.go
Comment (marckhouzam) [inline pkg/action/upgrade.go:239]: I don't mean to be a stickler on terminology but IIUC a "deprecated" API won't cause a problem but a "removed" API (which was deprecated in the previous versions of kube) will cause the problem.

So the work message should probably replace "deprecated" with "removed".

Comment (hickeyma) [reply pkg/action/upgrade.go]: Good point. Updated.

**Comment by marckhouzam** [comment]
I don't have the setup to test this, but the code looks good.  Nice improvement for this annoying situation.  Thanks!

LGTM

**Review by marckhouzam — APPROVED** [review]

**Comment by hickeyma** [comment]
Thanks for the reviews @Nastradamus and @marckhouzam. I will ask some of the other @helm/helm-core-maintainers to take a look at it also, as it is a bit of a workaround.

**Review by technosophos — APPROVED** [review]
Since it's a small bug, you technically didn't need a second review. However, I'm happy to review and say LGTM

**Comment by hickeyma** [comment]
Thanks @technosophos. Just wanted to make sure I wasn't violating any set standard in the community.

**Pull request merged**
